File: src/UserPageFragment/UserPageFragmentPagerAdapter.java
package UserPageFragment;

import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentStatePagerAdapter;

public class UserPageFragmentPagerAdapter extends FragmentStatePagerAdapter{

	public UserPageFragmentPagerAdapter(FragmentManager fm){
		super(fm);
	}

	@Override
	public Fragment getItem(int i){
		switch(i){
		case 0:
		default:
			return new _0_detail();
		case 1:
			return new _1_Tweet();
		case 2:
			return new _2_favorites();
		case 3:
			return new _3_follow();
		case 4:
			return new _4_follower();
		}
	}

	@Override
	public int getCount(){
		return 5;
	}

	@Override
	public CharSequence getPageTitle(int position){
		switch(position){
		case 0:
		default:
			return "Detail";
		case 1:
			return "Tweet";
		case 2:
			return "favorites";
		case 3:
			return "follow";
		case 4:
			return "follower";
		}
	}

}

File: app/src/main/java/sugtao4423/lod/ui/userpage/UserPageFragmentPagerAdapter.kt
package sugtao4423.lod.ui.userpage

import android.content.Context
import android.os.Bundle
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentStatePagerAdapter
import sugtao4423.lod.R
import sugtao4423.lod.ui.userpage.fragment.DetailFragment
import sugtao4423.lod.ui.userpage.fragment.StatusFragment
import sugtao4423.lod.ui.userpage.fragment.UserFragment

class UserPageFragmentPagerAdapter(fm: FragmentManager, private val context: Context) :
    FragmentStatePagerAdapter(fm, BEHAVIOR_RESUME_ONLY_CURRENT_FRAGMENT) {

    override fun getItem(i: Int): Fragment {
        return when (i) {
            1 -> StatusFragment().also {
                it.arguments = Bundle().apply {
                    putString(StatusFragment.KEY_FRAGMENT_TYPE, StatusFragment.TYPE_TWEET)
                }
            }
            2 -> StatusFragment().also {
                it.arguments = Bundle().apply {
                    putString(StatusFragment.KEY_FRAGMENT_TYPE, StatusFragment.TYPE_FAVORITE)
                }
            }
            3 -> UserFragment().also {
                it.arguments = Bundle().apply {
                    putString(UserFragment.KEY_FRAGMENT_TYPE, UserFragment.TYPE_FOLLOW)
                }
            }
            4 -> UserFragment().also {
                it.arguments = Bundle().apply {
                    putString(UserFragment.KEY_FRAGMENT_TYPE, UserFragment.TYPE_FOLLOWER)
                }
            }
            else -> DetailFragment()
        }
    }

    override fun getCount(): Int = 5

    override fun getPageTitle(position: Int): CharSequence {
        return when (position) {
            1 -> context.getString(R.string.page_title_tweet)
            2 -> context.getString(R.string.page_title_favorite)
            3 -> context.getString(R.string.page_title_follow)
            4 -> context.getString(R.string.page_title_follower)
            else -> context.getString(R.string.page_title_detail)
        }
    }

}
